ui-admin/src/study/notifications/SendTaskEmailModal.tsx

```diff
@@ -0,0 +1,149 @@
+      setTaskStatuses(['NEW'])
+    }
+    setTaskStatuses(statuses)
+  }
+
+
+  const [searchResults, setSearchResults] = useState<EnrolleeSearchExpressionResult[]>([])
+  useEffect(() => {
+    Api.findTriggersForStudyEnv(studyEnvContext.portal.shortcode, studyEnvContext.study.shortcode,
+      studyEnvContext.currentEnv.environmentName).then(result => {
+      setConfigs(result)
+      setIsLoading(false)
+    }).catch(() => {
+      Store.addNotification(failureNotification('Could not load notification configs'))
+    })
+  }, [])
+
+  const buildSearchExp = () => {
+    return concatSearchExpressions(taskStatuses.map(status => `{task.${taskStableId}.status}='${status}'`), 'or')
+  }
+
+  const searchExp = buildSearchExp()
+
+  useEffect(() => {
+    Api.executeSearchExpression(studyEnvContext.portal.shortcode, studyEnvContext.study.shortcode,
+      studyEnvContext.currentEnv.environmentName, searchExp).then(setSearchResults)
+  }, [searchExp])
+
+  const sendEmail = async () => {
+    if (!selectedConfig) {
+      return
+    }
+    try {
+      await Api.sendAdHocNotificationByEnrolleeSearchExp({
+        portalShortcode: studyEnvContext.portal.shortcode,
+        studyShortcode: studyEnvContext.study.shortcode,
+        envName: studyEnvContext.currentEnv.environmentName,
+        searchExpression: searchExp,
+        customMessages: { adHocMessage, adHocSubject },
+        triggerId: selectedConfig.id
+      })
+      Store.addNotification(successNotification('email processed'))
+    } catch (e) {
+      Store.addNotification(failureNotification('email processing failed'))
+    }
+    onClose()
+  }
+
+  const taskStatusOptions: { value: ParticipantTaskStatus, label: string}[] = [
+    { value: 'NEW', label: 'New' },
+    { value: 'IN_PROGRESS', label: 'In Progress' },
+    { value: 'COMPLETE', label: 'Complete' },
+    { value: 'REJECTED', label: 'Rejected' },
+    { value: 'VIEWED', label: 'Viewed' },
+    { value: 'REMOVED', label: 'Removed' }
+  ]
+
+
+  return <Modal onHide={onClose} show={true} className="modal-lg">
+    <Modal.Header closeButton>
+      <Modal.Title>Send Email To Enrollees Assigned {taskStableId}</Modal.Title>
+      <div className="ms-4">
+        {studyEnvContext.study.name}: {studyEnvContext.currentEnv.environmentName}
+      </div>
+    </Modal.Header>
+    <Modal.Body>
+      <form onSubmit={e => e.preventDefault()} className="py-3">
+        <div>
+          <label>
+            Send to enrollees with task status:
+            <Select
+              isMulti={true}
+              options={taskStatusOptions}
+              value={taskStatuses
+                .map(status =>
+                  taskStatusOptions
+                    .find(opt => opt.value === status))}
+              onChange={newVals => changeTaskStatus(
+                newVals
+                  ?.filter(newVal => !isNil(newVal))
+                  .map(newVal => newVal!.value))}/>
+          </label>
+        </div>
+
```

Comment:
this is a big batch of copy from AdHocEmailModal.   I'd just as soon centralize this into AdHocEmailModal by having it take an "assignedTaskStableId" option.  
That way it's clearer in code where all ad hoc emails come from, and later we could also build a task selector into that modal so you could also accomplish this directly from a general "send email to people" modal.

Reply:
Good idea - this reduced overall code. The modal is now a little bit on the complex side but the work to make it more modular seemed to be harder than would be worth it - if there end up being more recipient types, though, a refactor might be called for.